chainer/links/caffe/caffe_function.py

```diff
@@ -318,6 +319,54 @@ def _setup_relu(self, layer):
         self.forwards[layer.name] = fw
         self._add_layer(layer)
 
+    @_layer('Eltwise', 'ELTWISE')
+    def _setup_eltwise(self, layer):
+        operation = layer.eltwise_param.operation
+        coeffs = layer.eltwise_param.coeff
+        if coeffs == []:
+            coeffs = None
+
+        self.forwards[layer.name] = _EltwiseFunction(operation, coeffs)
+        self._add_layer(layer)
+
+    @_layer('Scale', 'NONE')    # TODO: old name
+    def _setup_scale(self, layer):
+        # Get layer parameters.
+        bottom = layer.bottom
+        blobs = layer.blobs
+        axis = layer.scale_param.axis
+        bias_term = layer.scale_param.bias_term
+
+        # Case of only bottom where W is learnt parameter.
+        if len(bottom) == 1:
+            W_blob = blobs[0]
+            bias_blob = blobs[1] if bias_term else None
+            func = _ScaleChain(axis, W_blob, bias_blob)
+        # Case of two bottoms where W is given as a bottom.
+        else:
+            bias_blob = blobs[0] if bias_term else None
+            func = _ScaleChain(axis, None, bias_blob)
+
+        self.add_link(layer.name, func)
+        self.forwards[layer.name] = _CallChildLink(self, layer.name)
+        self._add_layer(layer)
+
+    @_layer('Softmax', 'SOFTMAX')
+    def _setup_softmax(self, layer):
```

Comment:
Please add a comment that we only consider the situation in which input data is two dimensional and is taken softmax along last axis.